### PR TITLE
Tt 313 run the unit test before run the pre commit

### PR DESCRIPTION
**Pull request by DrFreud1**
### Observation
It would be better if we run unit tests before commiting, and only commit and push after tests are successful.

### Solution

To do so, on the package.json file it was needed just to modify the husky configuration as follows:
```
"husky": {
    "hooks": {
      "commit-msg": "commit-message-validator",
      "pre-commit": "ng lint && ng test --watch=false --browsers=ChromeHeadless"
    }
  }
```
**package-lock.json** is automatically generated for any operations where npm modifies either the node_modules tree, or **package.json**. It describes the exact tree that was generated, such that subsequent installs are able to generate identical trees, regardless of intermediate dependency updates.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -100,7 +100,7 @@
   "husky": {
     "hooks": {
       "commit-msg": "commit-message-validator",
-      "pre-commit": "ng lint"
+      "pre-commit": "ng lint && ng test --watch=false"
```
Comment (scastillo-jp) [inline package.json:103]: Parece que no esta funcionando muy bien.

Dañe un par de tests y en la hook solo detecta el  **lint** pero no los test

`"pre-commit": "ng lint && ng test --watch=false"
`
 hice la prueba cambiando el  orden

`"pre-commit": "ng test --watch=false && ng lint"
`y solo detecta  los test y no el lint.

como recomendación sería bueno que uses en los test  --browsers=ChromeHeadless
algo así
`"pre-commit": "ng test --watch=false --browsers=ChromeHeadless & ng lint" `

Comment (DrFreud1) [reply package.json]: Thank you for your comments, Sandro. It worked when I was testing the changes on my computer, maybe the problem was that I did not commit the package-lock.json file which I know now that automatically changes when changing the package.json file.

I have tested it again, and it works as expected. It does not allow you to perform the commit if a test fails or if the lint fails.

File: package.json [hunk 2]
```diff
@@ -80,7 +80,7 @@
     "commit-message-validator": "0.1.11",
     "datatables.net": "1.10.21",
     "datatables.net-dt": "1.10.21",
-    "husky": "4.2.3",
+    "husky": "^4.3.8",
```
Comment (ridouku) [inline package.json:83]: remove the caret symbol

**Review by ridouku — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (DrFreud1) [reply package.json:83]: Solved

**Review by ridouku — APPROVED** [review]

**Pull request closed without merging**